### PR TITLE
Remove Lerna and related root npm run scripts, fix types in rtm-api

### PR DESCRIPTION
**Pull request by filmaj**
### TODO

- [ ] update the required checks on this repo's settings and/or the versions of node we run tests against in this PR

###  Summary

Lerna, which is, or at least was, a monorepo management package, is now a few major versions ahead and its feature set has changed drastically. I do not believe it is suitable for our use case in this repo anymore. The commands we use it for are no longer supported in the newer versions.

Additionally, for the rtm-api package, depending on which version of the `@types/node` dependency you chanced upon retrieving, you may get a build error! For me, for example, I had no issues building the package as-is if I used `@types/node@20.4.9`. But, GitHub CI was using `@types/node@20.5.3` (see for example [this build](https://github.com/filmaj/node-slack-sdk/actions/runs/5955355744/job/16153831621) on my fork). I updated the type for one of the properties in this package to pass this failing build.

I updated the GitHub Actions workflow to not rely on Lerna anymore and simply iterate into each non-deprecated package, install its dependencies and run the tests. I also updated the workflow to only run tests against node v18 and v20, as v16 will be EOL'ed in 2 weeks.

Further discussion should be had about updating all of these packages' `engines` field and start moving up the minimum supported node version and releasing new major versions (I think?) as a result.

As an alternative, I tried using npm workspaces but encountered several issues with the `npm` CLI, the most serious of which was that `test` runs within sub-packages that errored would not cause `npm` to exit with a non-zero exit code (which would be reported as a success on GitHub Actions).

### CODE REVIEW

File: .github/workflows/ci-build.yml
Comment (filmaj) [inline .github/workflows/ci-build.yml:28]: Running `npm install` automatically runs the typescript builds ( 🤢 ) so in case this fails, which can happen very easily if your typescript dependencies drift or are different than on your own machine, this line will run `npm list` to output all installed dependencies and their versions. This helped me identify the dependency difference I was experiencing and why I had no trouble running builds on my local machine but GitHub CI did have errors doing so.

**Review by zimeg — APPROVED** [review]
Nice, this feels so much cleaner with staying up-to-date on all these versions! :raised_hands:

I agree that there's more discussion to have around bumping the supported Node versions in `engine`, but the changes here seem just fine as is! 🛳️

File: .github/workflows/ci-build.yml
Comment (filmaj) [reply .github/workflows/ci-build.yml:26]: Ooo those are great ideas! I will play around with that.

File: package.json
Comment (filmaj) [reply package.json:6]: The reference docs are actually generated manually according to the [maintainer's guide](https://github.com/slackapi/node-slack-sdk/blob/main/.github/maintainers_guide.md#-generating-documentation) (bottom of this section).

The package that does the generation under the `support/sdk-ref-docs` doesn't work on node 20 so I think this whole area needs a bunch more work 😓 and these docs haven't been updated in a long time it seems

File: .github/workflows/ci-build.yml
Comment (filmaj) [reply .github/workflows/ci-build.yml:26]: @zimeg take a look at this PR: https://github.com/filmaj/node-slack-sdk/pull/3
It implements your suggestion. The workflow file looks much nicer but what do you think about the 14 status checks now posted to the PR? I don't mind it and at a glance you can see exactly which project is passing/failing.

Comment (filmaj) [reply .github/workflows/ci-build.yml:26]: Another nice thing about this approach is that it runs each matrix combination in parallel, so feedback is faster.

Comment (zimeg) [reply .github/workflows/ci-build.yml:26]: Nice! Thanks for experimenting with this, I didn't expect that 14 statuses would appear!

It's a bit busier, but I think it's pretty nice that statuses for each package and version are shown separately. The parallel running is sweet too! I'm definitely a fan, but your call on making the change.

File: package.json
Comment (zimeg) [reply package.json:6]: Good to know! I think updating this section and doc generation in general can be addressed in another PR, so no need to worry about it here!

**Review by seratch — APPROVED** [review]
You may update some parts in the ci-build.yml before merging this, but overall it already looks good to me 👍

File: .github/workflows/ci-build.yml
Comment (WilliamBergamin) [reply .github/workflows/ci-build.yml:26]: :two-cents: I think that the 14 statuses provides good visual feedback to figure out what versions and packages are failing, also big fan of parallel jobs since it enables us to fail faster

**Pull request merged**
